app/ui/lib/Combobox.tsx

```diff
@@ -67,79 +85,83 @@ export const Combobox = ({
   onChange,
   onInputChange,
   allowArbitraryValues = false,
-  ariaLabel,
   hideOptionalTag,
+  ...props
 }: ComboboxProps) => {
-  const [query, setQuery] = useState(selected || '')
-
-  const q = query.toLowerCase()
+  const [query, setQuery] = useState(selectedItemValue || '')
+  const q = query.toLowerCase().replace(/\s*/g, '')
   const filteredItems = matchSorter(items, q, {
-    keys: ['value'],
+    keys: ['selectedLabel', 'label'],
     sorter: (items) => items, // preserve original order, don't sort by match
   })
-
   const zIndex = usePopoverZIndex()
-
+  const id = useId()
   return (
     <>
       <HCombobox
-        value={selected}
+        // necessary, as the displayed "value" is not the same as the actual selected item's *value*
+        value={selectedItemValue}
         // fallback to '' allows clearing field to work
         onChange={(val) => onChange(val || '')}
         onClose={() => setQuery('')}
-        defaultValue={selected}
         disabled={disabled || isLoading}
+        immediate
+        {...props}
       >
         {label && (
           // TODO: FieldLabel needs a real ID
           <div className="mb-2">
-            <FieldLabel id="FieldLabel" as="div" optional={!required && !hideOptionalTag}>
-              <Label>{label}</Label>
+            <FieldLabel
+              id={`${id}-label`}
+              htmlFor={`${id}-input`}
+              optional={!required && !hideOptionalTag}
+            >
+              {label}
             </FieldLabel>
-            {description && <TextInputHint id="TextInputHint">{description}</TextInputHint>}
+            {description && (
+              <TextInputHint id={`${id}-help-text`}>{description}</TextInputHint>
+            )}
           </div>
         )}
-        <ComboboxButton
-          className={cn(
-            `flex h-10 w-full grow items-center justify-between rounded border text-sans-md ring-accent-secondary data-[open]:ring-2`,
-            hasError
-              ? 'focus-error border-error-secondary hover:border-error'
-              : 'border-default hover:border-hover',
-            hasError && 'data-[open]:ring-error-secondary',
-            disabled
-              ? 'cursor-not-allowed text-disabled bg-disabled !border-default'
-              : 'bg-default',
-            disabled && hasError && '!border-error-secondary'
```

Comment:
I think some of these error classes need to be put back — red border and focus ring are missing.

<img width="558" alt="image" src="https://github.com/user-attachments/assets/bd5368d6-4c34-482b-9f9d-b4fe373d9041">

Reply:
As of [the latest commit](https://github.com/oxidecomputer/console/pull/2474/commits/d610bc3d8a6f3ab8e95d7c3ac136b001b147cb77):
<img width="530" alt="Screenshot 2024-10-03 at 1 05 32 PM" src="https://github.com/user-attachments/assets/0865953b-b7e5-4bf2-96b9-a6d201c898b4">